GameEngine/GameEngine.h

```diff
@@ -45,7 +46,7 @@ class GameState {
 };
 
 // The GameEngine class manages the game states and processes commands.
-class GameEngine {
+class GameEngine : public Map{
```

Comment:
this complicates things way too last minute i doubt you tested extensively plz leave how it was before